tests/src/core/testqgsexpression.cpp

```diff
@@ -1403,6 +1403,8 @@ class TestQgsExpression: public QObject
       QTest::newRow( "regexp_replace non greedy" ) << "regexp_replace('HeLLo','(?<=H).*?L', '-')" << false << QVariant( "H-Lo" );
       QTest::newRow( "regexp_replace cap group" ) << "regexp_replace('HeLLo','(eL)', 'x\\\\1x')" << false << QVariant( "HxeLxLo" );
       QTest::newRow( "regexp_replace invalid" ) << "regexp_replace('HeLLo','[[[', '-')" << true << QVariant();
+      QTest::newRow( "regexp_replace match, with null" ) << "regexp_replace('aaaa','a',NULL)" << false << QVariant( "" );
+      QTest::newRow( "regexp_replace no match with null" ) << "regexp_replace('zzz', 'b', NULL)" << false << QVariant( "zzz" ); // fix https://github.com/qgis/QGIS/issues/44274
```

Comment:
Can you add one for `regexp_replace('aaabbb', 'b', NULL)` as well?

Reply:
done.

Reply:
Thanks, I'd say this should also be NULL (`QVariant()`), just like `"" + NULL ==> NULL`